### PR TITLE
[BH-2034] Various cleanups and refactors across the OS

### PR DESCRIPTION
**Pull request by Lefucjusz**
* Removed some legacy files that were compiled, 
but not used anywhere.
* Cleaned-up ImageManager and FontManager 
implementations, trying to use RAII wherever 
possible.
* Fixed minor bugs in Font.cpp implementation.
* Removed unused includes.
* Other minor cleanups and refactors.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [ ] Has changelog entry added

### CODE REVIEW

**Review by Maciej-Mudita — APPROVED** [review]

**Review by dawidwojtas-mudita — APPROVED** [review]
LGTM ;)

File: module-gui/gui/core/PixMap.cpp [hunk 1]
```diff
@@ -1,59 +1,48 @@
...
-
-        type = Type::PIXMAP;
+        name   = {};
+        type   = Type::Pixmap;
     }
 
-    PixMap::PixMap(uint16_t w, uint16_t h, uint8_t *data) : ImageMap(w, h, data)
+    PixMap::PixMap(std::uint16_t w, std::uint16_t h, std::uint8_t *pixMapData) : ImageMap(w, h, pixMapData)
     {
+        data = new std::uint8_t[width * height];
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply module-gui/gui/core/PixMap.cpp:19]: I left it intentionally, as this allocation is done in ctor, so there's no way to return any value - IMO the cleanest way is to let this allocation just throw an exception if something goes wrong

File: module-gui/gui/core/VecMap.cpp [hunk 2]
```diff
@@ -1,53 +1,51 @@
...
-        type = Type::VECMAP;
+        name   = {};
+        type   = Type::Vecmap;
     }
 
-    VecMap::VecMap(uint16_t w, uint16_t h, uint8_t *data) : ImageMap(w, h, data)
+    VecMap::VecMap(std::uint16_t w, std::uint16_t h, std::uint8_t *vecMapData) : ImageMap(w, h, vecMapData)
     {
+        // Add empty vectors for all rows
+        data = new std::uint8_t[height];
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply module-gui/gui/core/VecMap.cpp:20]: I left it for the same reason as in `PixMap`

File: module-gui/gui/core/RawFont.cpp [hunk 3]
```diff
@@ -19,65 +19,70 @@ namespace gui
...
         std::uint32_t offset = 0;
 
         if (info.load(data, offset) != gui::Status::GUI_SUCCESS) {
             return gui::Status::GUI_FAILURE;
         }
 
-        // number of glyphs in the font
-        memcpy(&glyph_count, data + offset, sizeof(std::uint32_t));
+        // Number of glyphs in the font
+        std::memcpy(&glyph_count, &data[offset], sizeof(std::uint32_t));
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply module-gui/gui/core/RawFont.cpp]: Done

File: products/BellHybrid/apps/common/include/common/models/UserSessionModel.hpp [hunk 4]
```diff
@@ -53,7 +53,7 @@ namespace app
 
       private:
         SessionState sessionState = SessionState::Inactive;
-        sys::Service *serv;
+        sys::Service *serv{nullptr};
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply products/BellHybrid/apps/common/include/common/models/UserSessionModel.hpp]: Done

**Pull request merged**